### PR TITLE
[IMP] base_location_geonames_import

### PR DESCRIPTION
**Pull request by pedrobaeza**
* Speed improvement applying cache for not doing duplicated searches over states.
* Tests changed to SavepointCase, which only passes over setUp one time, saving test time.

With Spanish toponyms, I have down the time from 5'30" to 4'20"

cc @Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@alexis-via @Yajo, please review

File: base_location_geonames_import/wizard/geonames_import.py [hunk 1]
```diff
@@ -89,25 +89,27 @@ def create_better_zip(self, row, country):
...
-            raise UserError(
-                _("Too many states with code %s for country %s")
-                % (row[code_row_index], country.code))
-        if len(states) == 1:
-            return states[0]
+            ('country_id', '=', country_id),
+            ('code', '=', code),
+        ])
+        if len(states):
+            return states[:-1]
```
Comment (yajo) [inline base_location_geonames_import/wizard/geonames_import.py]: It is cleaner and more efficient to pass `limit=1` to the `search` method.

**Review by yajo — APPROVED** [review]
Code review, just one little improvement suggestion.

File: base_location_geonames_import/wizard/geonames_import.py (hunk 1 above)
Comment (pedrobaeza) [reply base_location_geonames_import/wizard/geonames_import.py]: Yeah, indeed

**Comment by yajo** [comment]
Seems like you lost a little coverage, not so hard to achieve.

**Comment by pedrobaeza** [comment]
@carlos-dauden, can you review please?

**Review by petrus-v — APPROVED** [review]
LGTP :+1:

**Pull request merged**